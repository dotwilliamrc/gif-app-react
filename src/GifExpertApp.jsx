import { useState } from 'react'
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([])

  const handleCategories = (value) => {
    const cat = categories.filter(v => v !== value)
    setCategories([value, ...cat])
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        // setCategories={ setCategories }
        onNewCategory={ (value) => handleCategories(value) }
      />
      { categories.map(category => <GifGrid key={ category } category={ category } />) }
    </>
  )
}
