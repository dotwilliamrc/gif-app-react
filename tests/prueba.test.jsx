import { render, container } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'

describe('Esta es una prueba de la prueda', () => {
  test('primera prueba', () => {
    render(<GifExpertApp />)

    expect(container).toMatchSnapshot()
  })
})
