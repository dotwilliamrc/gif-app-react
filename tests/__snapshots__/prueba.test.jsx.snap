// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Esta es una prueba de la prueda primera prueba 1`] = `undefined`;
